### PR TITLE
#14730 - Add tests for findFirst() with primary key as a string

### PR DESCRIPTION
**Pull request by Jeckerson**
Hello!

* Type: bug fix
* Link to issue: #14730

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Related - phalcon/docs#2647

Thanks

### CODE REVIEW

File: phalcon/Mvc/Model/Query/Builder.zep [hunk 1]
```diff
@@ -633,17 +638,7 @@ class Builder implements BuilderInterface, InjectionAwareInterface
              * If the conditions is a single numeric field. We internally create
              * a condition using the related primary key
              */
-            if typeof models == "array" {
-                if unlikely count(models) > 1 {
-                    throw new Exception(
-                        "Cannot build the query. Invalid condition"
-                    );
-                }
```
Comment (sergeyklay) [inline phalcon/Mvc/Model/Query/Builder.zep]: Could you please explain this removal in the PR description and mention this in the change log

File: phalcon/Mvc/Model/Query/Builder.zep [hunk 2]
```diff
@@ -1687,4 +1681,20 @@ class Builder implements BuilderInterface, InjectionAwareInterface
 
         return this;
     }
+
+    /**
+     * Get first key from models array
+     */
+    private function getFirstModel(var models) -> string
```
Comment (sergeyklay) [inline phalcon/Mvc/Model/Query/Builder.zep]: Do we really need a separated method for if/else which will be used once?

File: tests/integration/Mvc/ModelCest.php [hunk 3]
```diff
@@ -51,6 +54,33 @@ public function executeCamelCaseRelation(IntegrationTester $I)
...
+     * @param Example $example
+     */
+    public function findFirstByStringPrimaryKey(IntegrationTester $I, Example $example)
+    {
+        $I->wantToTest('Phalcon\Mvc\Model - findFirst()');
+
+        (new StringPrimaryMigration())($this->container->get('db'));
+        $model = ModelWithStringPrimary::findFirst($example['params']);
+
+        $I->assertSame($example['found'], $model instanceof Model);
```
Comment (sergeyklay) [inline tests/integration/Mvc/ModelCest.php]: AFAIK, `assertSame` compare by value but `assertEquals` compares identity. Thus,
```php
$I->assertSame(TRUE, 1); //  TRUE
```

IF I'm not wrong, this should use assertEquals, or better:
```php
$I->assertTrue($example['expected'] === $model instanceof Model);
```

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: phalcon/Mvc/Model/Query/Builder.zep (hunk 1 above)
Comment (Jeckerson) [reply phalcon/Mvc/Model/Query/Builder.zep]: It already make same check above

https://github.com/phalcon/cphalcon/pull/14758/files#diff-6801ddb1a4b13b44acd35b479d000f4bR620-R625

File: tests/integration/Mvc/ModelCest.php (hunk 3 above)
Comment (Jeckerson) [reply tests/integration/Mvc/ModelCest.php]: Nope, it is vice versa.

'2204' !== 2204
assertSame('2204', 2204) // this test fails

'2204' == 2204
assertEquals('2204', 2204) // this test passes

File: phalcon/Mvc/Model/Query/Builder.zep (hunk 2 above)
Comment (Jeckerson) [reply phalcon/Mvc/Model/Query/Builder.zep]: Used in previous iteration for DRY approach. Decided to leave it here, to make `getPhql()` little bit lighter of code.

File: phalcon/Mvc/Model/Query/Builder.zep (hunk 1 above)
Comment (sergeyklay) [reply phalcon/Mvc/Model/Query/Builder.zep]: No.
```diff
-   if unlikely !count(models)
+   if unlikely count(models) > 1
```

File: phalcon/Mvc/Model/Query/Builder.zep (hunk 2 above)
Comment (sergeyklay) [reply phalcon/Mvc/Model/Query/Builder.zep]: And to make Phalcon slower I guess )

File: tests/integration/Mvc/ModelCest.php (hunk 3 above)
Comment (sergeyklay) [reply tests/integration/Mvc/ModelCest.php]: Okay, thank you

File: phalcon/Mvc/Model/Query/Builder.zep (hunk 1 above)
Comment (Jeckerson) [reply phalcon/Mvc/Model/Query/Builder.zep]: I doubt if this really needed here, but will revert anyway.

**Comment by Jeckerson** [comment]
Related - phalcon/docs#2647

**Review by ruudboon — APPROVED** [review]

**Review by sergeyklay — APPROVED** [review]
Looks good to me.

**Comment by sergeyklay** [comment]
Btw, PR title doesn't reflect actual changes

**Comment by Jeckerson** [comment]
Title changed

**Review by niden — APPROVED** [review]

**Pull request merged**